types/auth0/index.d.ts

```diff
@@ -1538,8 +1538,8 @@ export class ManagementClient<A = AppMetadata, U = UserMetadata> {
     createDevicePublicKey(data: Data): Promise<User<A, U>>;
     createDevicePublicKey(data: Data, cb: (err: Error, data: any) => void): void;
 
-    deleteDeviceCredential(params: ClientParams): Promise<User<A, U>>;
-    deleteDeviceCredential(params: ClientParams, cb: (err: Error, data: any) => void): void;
+    deleteDeviceCredential(params: ObjectWithId): Promise<void>;
```

Comment:
what is object with id? if anything, should it be something self-describing? (DeleteParams),
they call it `CredentialParameters`-type IMO:
https://github.com/auth0/node-auth0/blob/08d23024e4051516bd2fbb8c8ab94d4b8ef42c9a/src/management/DeviceCredentialsManager.js#L103

Reply:
as this is one-prop object, we could just inline this type and avoid naming problems altogether

Reply:
`ObjectWithId` is defined as
```
export interface ObjectWithId {
    id: string;
}
```
I tried to follow the patterns I observed in the existing definitions and `ObjectWithId` is used throughout the definition (92 times in total) whenever a params object only contains an `id` field.

Reply:
The named type is more consistent with the rest of the api even if it feels like using an existing type for so many different things. If it needs to diverge for a future change, the inline one would need to change anyway. I'm ok with the named ubiquitous one.

Reply:
Yes, I;m sorry, I missed `ObjectWithId` in other places.